examples/lazy-reader-tests/src/types_api2_mol2.rs

```diff
@@ -0,0 +1,796 @@
+use super::TypesCheckErr;
+use crate::types_api;
+use crate::types_api2::*;
+use molecule::lazy_reader::Cursor;
+
+pub trait Mol2Vec {
+    type RetType;
+    fn mol_len(&self) -> Result<usize, TypesCheckErr>;
+    fn mol_get(&self, index: usize) -> Result<Self::RetType, TypesCheckErr>;
+}
+impl Mol2Vec for Vec<u8> {
```

Comment:
I'd like to know how the test code is generated, because it doesn't look like it's handwritten.

Reply:
it's handwritten